File: tinylog/src/main/java/com/androidwind/log/ILogConfig.java
package com.androidwind.log;

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
public interface ILogConfig {
    ILogConfig setEnable(boolean isEnable);

    ILogConfig setWritable(boolean writable);

    ILogConfig setLogPath(String logPath);

    ILogConfig setFileSize(int fileSize);

    ILogConfig setLogCallBack(LogConfig.LogCallBack callBack);

    void apply();
}

File: tinylog-kotlin/src/main/java/com/androidwind/log/ILogConfig.kt
package com.androidwind.log

/**
 * interface to define the implementation class's function
 *
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
interface ILogConfig {
    fun setEnable(isEnable: Boolean): ILogConfig

    fun setWritable(writable: Boolean): ILogConfig

    fun setLogPath(logPath: String): ILogConfig

    fun setFileSize(fileSize: Int): ILogConfig

    fun setLogCallBack(callBack: LogConfig.LogCallBack): ILogConfig

    fun setEncrypt(key: String): ILogConfig

    fun setLogLevel(level: Int): ILogConfig

    fun apply()
}
